tests/core/_scheduler/test_scheduler.py

```diff
@@ -472,7 +482,6 @@ def modified_config_task(n):
 
 def test_can_exec_task_with_modified_config():
```

Comment:
@trgiangdo @jrobinAV 
Do you think this test is still relevant in this PR? Maybe what we should test here is can't modify during task execution?

Reply:
I don't get your idea, but I think it's not relevant to this PR anyway.
The test is only modified so that we only build the dispatcher after all configurations.

Reply:
oh I think the idea of this test is to see if we can modify the config during job execution (meaning a dispatcher has already been built and I'm assuming it's something we don't want to do). So the purpose of this test is contradicting with the PR itself

Reply:
Oh, it's passing, which is not good.

So the problem is that we cannot modify the config by `Config.config_global_app()`, but by directly assign using `Config.global_config.property = ...` is still allowed. I need to fix this.

Reply:
Exactly! :D

Reply:
Actually, the idea of this test is to check in stand-alone mode, if we modify the `global_config` before submitting the task, will the `global_config` update? There is no setter here.

I'll create a new test for this purpose.

Reply:
In standalone mode, task execution is performed in a dedicated subprocess that does not share the memory with the main process. That means the Config singleton is not shared. Each subprocess builds another instance of the Config. For that purpose, we serialize the config as a string in the main process and pass it as a parameter to the subprocess. Then at the very beginning of the subprocess task execution, we rebuild a new Config using the string parameter.

The purpose of this test is to ensure that each subprocess takes into account the Config modifications done in the main Process when it builds a new instance of the Config.

With your new implementation, the _ConfigBlocker is a new Singleton that is not passed to the subprocesses. That means in each subprocess, the config will not be blocked. After rebuilding a new Config in each subprocess, we should block the Config. [See line 138 of file _job_dispatcher.py](https://github.com/Avaiga/taipy-core/blob/develop/src/taipy/core/_scheduler/_dispatcher/_job_dispatcher.py#L138)

Reply:
May be this explanation should be added as a comment in the test to explain the purpose.

Reply:
This is exactly the problem that I'm facing right now. Thank you 😆

Reply:
I've added a test for this.

Since the exception when running task is caught by the logger, I cannot check for the exception, so I add a `pytest.raises` directly inside the task function.